packages/headless/src/features/history/history-slice.ts

```diff
@@ -47,29 +47,64 @@ const isEqual = (current: HistoryState, next: HistoryState) => {
 const isContextEqual = (current: ContextState, next: ContextState) =>
   JSON.stringify(current.contextValues) === JSON.stringify(next.contextValues);
 
-// TODO: compare all facets non-idle values, comparing current values changes history after 1st request
-const isFacetsEqual = (current: FacetSetState, next: FacetSetState) =>
-  JSON.stringify(current) === JSON.stringify(next);
+type FacetStateWithCurrentValues = Record<
+  string,
+  CurrentValues<BaseFacetValueRequest>
+>;
 
-const isDateFacetsEqual = (
-  current: DateFacetSetState,
-  next: DateFacetSetState
-) => JSON.stringify(current) === JSON.stringify(next);
+const isFacetsEqual = (
+  current: FacetStateWithCurrentValues,
+  next: FacetStateWithCurrentValues
+) => {
+  for (const [key, value] of Object.entries(next)) {
```

Comment:
Is it important to consider that case where `current` has a value that is not present in `next` (e.g. selecting a facet value in one changes the values of another)? I guess it would be caught as a change in another facet, I'm not sure?

Reply:
To me that's not a strong indicator of history change. 
> selecting a facet value in one changes the values of another
The selection is the strong indicator here

Reply:
Reminder that this feature here, the isEqual, only serves to compare 2 states, it doesn't not indicate what is part of the snapshot or not